pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -561,6 +568,8 @@ func getLocalIP() ([]v1.NodeAddress, error) {
 									Address: ipnet.IP.String(),
 								},
 							)
+						} else {
+							klog.Warningf("Failed to patch IP as MAC address %q does not belong to a VMware platform", i.HardwareAddr.String())
 						}
 						klog.V(4).Infof("Find local IP address %v and set type to %v", ipnet.IP.String(), addressType)
```

Comment:
Can we fix this log line? I don't think it should log if the prefix doesn't match, also `addressType` will always be empty.

Reply:
@andrewsykim @frapposelli:

Let's move this log message into the conditional after the addresses are. For example:

```go
v1.NodeAddress{
  Type:    v1.NodeInternalIP,
  Address: ipnet.IP.String(),
  },
)

klog.V(4).Infof("Detected local IP address as %v.", ipnet.IP.String())

} else {

klog.Warningf("Failed to patch IP as MAC address %q does not belong to a VMware platform", i.HardwareAddr.String())
						}
						
					}
```

And remove `addressType`. Thoughts?

Reply:
should be fixed now